loadsbroker/aws.py

```diff
@@ -279,23 +276,23 @@ def update_state(inst):
         while time.time() < end_time and pending:
             self.debug('%d pending instances.' % len(pending))
             # Update the state of all the pending instances
-            yield [self.execute(update_state, inst) for inst in pending]
+            await gen.multi(
+                [self.execute(update_state, inst) for inst in pending])
             pending = self.pending_instances()
 
             # Wait if there's pending to check again
             if pending:
-                yield self.wait(interval)
+                await self.wait(interval)
 
         # Remove everything that isn't running by now
         dead = self.dead_instances() + self.pending_instances()
 
         # Don't wait for the future that kills them
         self.debug("Removing %d dead instances that wouldn't run" % len(dead))
-        self.remove_instances(dead)
+        gen.convert_yielded(self.remove_instances(dead))
```

Comment:
I'm actually confused why I didn't originally yield from remove_instances, since its a coroutine. Calling it like I did never actually ran it, so the code was broken the whole time.

Since remove_instances is a coroutine, shouldn't this be an await for it?